libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts

```diff
@@ -227,10 +227,12 @@ export class GoFeatureFlagProvider implements Provider {
     const sdkResponse: ResolutionDetails<T> = {
       value: apiResponseData.value,
       variant: apiResponseData.variationType,
-      reason: apiResponseData.reason.toString(),
+      reason: apiResponseData.reason?.toString() || 'UNKNOWN'
     };
-    if (apiResponseData.errorCode) {
-      sdkResponse.errorCode = apiResponseData.errorCode.toString();
+    if (Object.values(ErrorCode).includes(apiResponseData.errorCode as ErrorCode)) {
+      sdkResponse.errorCode = ErrorCode[apiResponseData.errorCode as ErrorCode];
+    } else if (apiResponseData.errorCode) {
+      sdkResponse.errorCode = ErrorCode.GENERAL;
```

Comment:
This is probably not going to happen, but it looks like the existing code would return error codes that goff sent it the HTTP response.

Now, those errors code are an enum, so I have to do some mapping and if I can't map it, I use `GENERAL`.

Reply:
Yes it should probably throw in this case let me look at that in another PR if you want.

Reply:
Ya, I think it's a but unexpected, but this is about your REST API so it's up to you! to decide exactly how this should behave.